### ISSUE TITLE
Feature request: Add conditional expressions and loops (if/for) support to Nuxt MDC

### ISSUE THREAD

**Issue by maximepvrt** [nuxt-modules/mdc#43]
### Issue Description
Nuxt MDC is a fantastic tool for enhancing Markdown with dynamic content through the use of variables. However, it currently lacks support for conditional expressions (using 'if') and loops (using 'for'). These features are crucial for creating more dynamic and complex content within Markdown documents. 

### Feature Request
I request the addition of support for conditional expressions and loops in Nuxt MDC. This would greatly enhance its capabilities and make it even more versatile for various use cases.

### Proposed Implementation
To implement these features, we could introduce a syntax similar to JavaScript's 'if' and 'for' statements within Markdown code blocks. For example:

#### Conditional Expression (if)
```markdown
{if condition}
   This content will be displayed if the condition is true.
{else}
   This content will be displayed if the condition is false.
{/if}
```

### Loop (for)
```markdown
{for item in array}
   This content will be repeated for each item in the array.
{/for}
```

### Benefits
The addition of conditional expressions and loops would open up new possibilities for using Nuxt MDC, such as creating dynamic lists, conditionally rendering content, and more. This enhancement would make the tool even more powerful and flexible.

**Comment by nobkd** [comment]
So you want to include some more features into MDC and Markdown, that are common in templating languages like [Handlebars](https://handlebarsjs.com/guide/#evaluation-context), [Nunjucks](https://mozilla.github.io/nunjucks/templating.html#if) and [Liquid](https://liquidjs.com/tutorials/intro-to-liquid.html#Tags), right?

**Comment by maximepvrt** [comment]
Yes, currently mdc supports binding, next step is to add conditions and loops

**Comment by maximepvrt** [comment]
@farnabaz, would you be willing to add this syntax? I'm open to trying it out with Micromark.

**Comment by adamdehaven** [comment]
If we had a way to pass data to a default slot that could be parsed in the MDC and passed along, this would almost already work; essentially, something like what I have below:

---

First, you would create a block component and globally register it (I'm using Nuxt for this example, but all the same).

```vue
// ContainerLoop.vue
<template>
  <slot
    v-for="item in data"
    :key="item.id"
    :item="item"
  />
</template>

<script setup lang="ts">
import type { PropType } from 'vue'

defineProps({
  /** The iterator data that will be passed in to the slot content */
  data: {
    type: Array as PropType<Record<string, any>[]>,
    required: true,
  },
})
</script>
```

Then in the MDC document, you could do something like this:

```md
---
features:
  - id: 1
    title: Feature One
    description: "ONE: This is the content of my feature."
  - id: 2
    title: Feature Two
    description: "TWO: This is the content of my feature."
---

::container-loop{:data="$doc.features"}
  Hello {{ $item.title }}
::
```

@farnabaz am I missing anything internal that would make this work as-is? I'd be willing to

**Comment by adamdehaven** [comment]
I'm fairly positive the data needs to be injected into [here](https://github.com/nuxt-modules/mdc/blob/b89db73a82624760cd6fa5b85ede263b95ff157e/src/runtime/components/MDCRenderer.vue#L170-L174) when the specific component is processed but can't seem to get it working 🤔

**Comment by adamdehaven** [comment]
Related issue in `remark-mdc`: https://github.com/nuxtlabs/remark-mdc/issues/69
